lib/internal/util/inspect.js

```diff
@@ -593,10 +597,26 @@ function isInstanceof(object, proto) {
   }
 }
 
+// Special-case for some builtin prototypes in case their `constructor` property has been tampered.
+const wellKnownPrototypes = new SafeMap();
+wellKnownPrototypes.set(ObjectPrototype, { name: 'Object', constructor: Object });
+wellKnownPrototypes.set(FunctionPrototype, { name: 'Function', constructor: Function });
```

Comment:
```suggestion
wellKnownPrototypes.set(ObjectPrototype, { __proto__: null, name: 'Object', constructor: Object });
wellKnownPrototypes.set(FunctionPrototype, { __proto__: null, name: 'Function', constructor: Function });
```
?

Reply:
Why?

Reply:
to avoid the future footgun of someone accessing a different property on these objects, or adding an object without specifying these exact two properties

Reply:
(isn't there a lint rule already that should be requiring this? i added one in https://github.com/nodejs/node/pull/46083)

Reply:
I'm not convinced any of those would ever happen, it looks like pure overhead to me, I'll skip this time